pkg/extension/sumologicextension/extension.go

```diff
@@ -39,6 +38,7 @@ type SumologicExtension struct {
 	baseUrl          string
 	conf             *Config
 	logger           *zap.Logger
+	authenticator    Authenticator
```

Comment:
This should be an interface for more generic implementation:

```suggestion
	authentication    Authentication
```